web/src/components/FeedbackButton.vue

```diff
@@ -1,25 +1,25 @@
-<script>
-export default {
-  props: {
-    front: {
-      type: Boolean,
-      default: false,
-    },
-  },
-  data() {
-    return {
-      visible: false,
-      form: 'https://docs.google.com/forms/d/e/1FAIpQLScWccZqTF_b6nQkfkAvLNf7XGV9k7Rqf8V_Mid-oZLY47g_IQ/viewform',
-      width: 640,
-      height: 930,
-    };
-  },
-  computed: {
-    src() {
-      return `${this.form}?embedded=true`;
-    },
-  },
-};
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator';
+
+@Component
+export default class FeedbackButton extends Vue {
+  @Prop({
+    default: false,
+  })
+  readonly front!: boolean;
+
+  visible = false;
+
+  readonly form = 'https://docs.google.com/forms/d/e/1FAIpQLScWccZqTF_b6nQkfkAvLNf7XGV9k7Rqf8V_Mid-oZLY47g_IQ/viewform';
+
+  readonly width = 640;
+
+  readonly height = 930;
```

Comment:
Is there a different way to express the data block?  This looks like how I express non-reactive properties in other places.  Might get used to it though.  Just curious.

Reply:
This is one place where I find the Vue.extend() way to be better (though that is due mainly to familiarity with it). Here's an [example from Multinet](https://github.com/multinet-app/multinet/blob/master/client/src/views/GraphDetail.vue#L204-L212). The `as` keyword has a natural idiomatic usage here in explicitly providing types for data props.

Reply:
Yeah I dig that syntax.  Ah well, might just be one of those small concessions.

Reply:
in this case the typings are not needed since for `width = 640` Typescript knows that is a number. for an array I would type `arr: number[] = [];`